src/Tribe/Views/V2/Widgets/Widget_List.php

```diff
@@ -78,28 +85,80 @@ class Widget_List extends Widget_Abstract {
 		'jsonld_enable'        => 'tribe_is_truthy',
 	];
 
-	/**
-	 * @todo update in TEC-3612 & TEC-3613
-	 *
-	 * {@inheritDoc}
-	 */
-	public function setup() {
-
-	}
-
 	/**
 	 * {@inheritDoc}
 	 */
-	public function get_arguments() {
+	public function get_arguments( $instance = [] ) {
 		$arguments = $this->arguments;
 
 		$arguments['description'] = esc_html__( 'A widget that displays upcoming events.', 'the-events-calendar' );
 		// @todo update name once this widget is ready to replace the existing list widget.
 		$arguments['name']                          = esc_html__( 'Events List V2', 'the-events-calendar' );
 		$arguments['widget_options']['description'] = esc_html__( 'A widget that displays upcoming events.', 'the-events-calendar' );
 
+		// Setup default title.
+		$arguments['title']        = __( 'Upcoming Events', 'the-events-calendar' );
+
+		// Setup admin fields.
+		$arguments['admin_fields'] = [
+			'title'                => [
+				'label' => __( 'Title:', 'the-events-calendar' ),
+				'type'  => 'text',
+			],
+			'limit'                => [
+				'label' => __( 'Show:', 'the-events-calendar' ),
+				'type'  => 'dropdown',
+				'options' => $this->get_limit_options(),
+			],
+			'no_upcoming_events'   => [
+				'label' => __( 'Show widget only if there are upcoming events', 'the-events-calendar' ),
+				'type'  => 'checkbox',
+			],
+			'featured_events_only' => [
+				'label' => _x( 'Limit to featured events only', 'events list widget setting', 'the-events-calendar' ),
+				'type'  => 'checkbox',
+			],
+			'jsonld_enable'        => [
+				'label' => __( 'Generate JSON-LD data', 'the-events-calendar' ),
+				'type'  => 'checkbox',
+			],
+
+		];
+
+		// Add the Widget to the arguments to pass to the admin template.
+		$arguments['widget_obj'] = $this;
+
 		$arguments = wp_parse_args( $arguments, $this->get_default_arguments() );
 
 		return $this->filter_arguments( $arguments );
 	}
+
+	/**
+	 *
+	 *
+	 * @since TBD
+	 *
+	 * @return array
+	 */
+	public function get_limit_options() {
+		/**
+		 * Filter the max limit of events to display in the List Widget.
+		 *
+		 * @since TBD
+		 *
+		 * @param int The max limit of events to display in the List Widget, default 10.
+		 */
+		$events_limit = apply_filters( "tribe_events_widget_list_events_max_limit", 10 );
```

Comment:
:no_entry_sign: String "tribe_events_widget_list_events_max_limit" does not require double quotes; use single quotes instead

<sub>hash: 937964f9103c7646d8503f0f131056d9</sub>